libs/sdk-ui-filters/src/DateFilter/DateFilterBody/DateFilterBody.tsx

```diff
@@ -397,57 +334,156 @@ export class DateFilterBody extends React.Component<IDateFilterBodyProps, IDateF
+                >
+                    {isEditMode && !isMobile && showHeaderMessage ? <EditModeMessage /> : null}
+                    {isMobile ? (
+                        renderMobileContent()
+                    ) : (
+                        <VisibleScrollbar className={visibleScrollbarClassName} style={scrollerStyle}>
+                            {renderDefaultContent()}
+                        </VisibleScrollbar>
+                    )}
+                </div>
+            </>
+        );
     };
-}
+
+    const {
+        isExcludeCurrentPeriodEnabled,
+        isMobile,
+        onApplyClick,
+        onCancelClick,
+        closeDropdown,
+        onConfigurationClick,
+        dateFilterButton,
+        errors,
+        isConfigurationEnabled,
+        withoutApply,
+    } = props;
+
+    const showExcludeCurrent: boolean = !isMobile || isExcludeCurrentPeriodEnabled;
+
+    return (
+        <div className="gd-extended-date-filter-container">
+            <div
+                className={cx(
+                    "gd-extended-date-filter-body",
+                    "s-extended-date-filters-body",
+                    isMobile && "gd-extended-date-filter-body-mobile",
+                )}
+            >
+                {route === null && isMobile ? (
+                    <div
+                        onClick={() => {
+                            onCancelClick();
+                            closeDropdown();
+                        }}
+                    >
+                        {dateFilterButton}
+                    </div>
+                ) : null}
+                {renderDateFilterBody()}
+                {showExcludeCurrent ? renderExcludeCurrent() : null}
+
+                <div className={cx("gd-extended-date-filter-actions")}>
+                    <div className="gd-extended-date-filter-actions-left-content">
+                        {isConfigurationEnabled ? (
+                            <DateFilterConfigurationButton onConfiguration={onConfigurationClick} />
+                        ) : null}
+                    </div>
+                    <div className="gd-extended-date-filter-actions-buttons">
+                        <DateFilterBodyButton
+                            messageId={withoutApply ? "close" : "cancel"}
+                            className="gd-button-secondary gd-button-small s-date-filter-cancel"
+                            onClick={() => {
+                                onCancelClick();
+                                closeDropdown();
+                            }}
+                        />
+                        {!withoutApply && (
+                            <DateFilterBodyButton
+                                id={DATE_FILTER_APPLY_BUTTON_ID}
+                                messageId="apply"
+                                className="gd-button-action gd-button-small s-date-filter-apply"
+                                disabled={!isEmpty(errors)}
```

Comment:
disabled state is defined only by errors, so we can use them exactly in keyboardHandler too. Or wrap to some helper to reuse it. Also consider that feature to submit all filters at once hides Apply button on individual filters and probably needs to also disable Submit in keyboard handler